src/main/java/br/unb/cic/analysis/ioa/InterproceduralOverrideAssignment.java

```diff
@@ -398,15 +398,18 @@ private void kill(FlowSet<DataFlowAbstraction> in, Unit unit) {
     }
 
     private void removeAll(ValueBox valueBox, FlowSet<DataFlowAbstraction> rightOrLeftList) {
+        FlowSet<DataFlowAbstraction> itemsToRemoved = new ArraySparseSet<>();
         rightOrLeftList.forEach(dataFlowAbstraction -> {
             try {
                 if (containsValue(dataFlowAbstraction, valueBox.getValue())) {
-                    rightOrLeftList.remove(dataFlowAbstraction);
+                    itemsToRemoved.add(dataFlowAbstraction);
                 }
             } catch (ValueNotHandledException e) {
                 e.printStackTrace();
             }
         });
+
+        rightOrLeftList.difference(itemsToRemoved);
```

Comment:
@barbosamaatheus qual o erro que dava? ou isso é apenas uma refatoração?

itemsToBeRemoved would be better

Reply:
Antes eu estava removendo o item da lista que estava percorrendo no for. Quando essa remoção acontecia os índices da lista eram modificados e isso ocasionava na não remoção de outros itens que deveriam ser removidos.